pkg/storegateway/blocks.gohtml

```diff
@@ -90,5 +111,82 @@
     {{ end }}
     </tbody>
 </table>
+<script>
+    // Get the dropdown and elements to control
+    var dropdown = document.getElementById("action-type");
+    var elementsToControl = document.querySelectorAll(".item-to-control");
+
+    // Function to toggle visibility for each element
+    function toggleVisibility(isVisible) {
+        elementsToControl.forEach(function (element) {
+            if (isVisible) {
+                element.removeAttribute("hidden");
+            } else {
+                element.setAttribute("hidden", "hidden");
+            }
+        });
+    }
+
+    // Add an event listener to the dropdown
+    dropdown.addEventListener("change", function () {
+        var isVisible = dropdown.value !== "none";
+        toggleVisibility(isVisible);
+    });
+
+    // Set the initial visibility based on the selected option
+    var isVisible = dropdown.value !== "none";
+    toggleVisibility(isVisible);
+
+    // Function to reload the page with the action
+    function reloadWithAction() {
```

Comment:
This function submits the form and performs the action, it doesn't just reload.

It should be called `submitForm` or maybe `submitFormAndReload` if you want to underline the fact that it will reload.